src/compiler/diagnosticMessages.json

```diff
@@ -763,7 +763,7 @@
         "category": "Error",
         "code": 1239
     },
-    "Unable to resolve signature of property decorator when called as an expression.": {
+    "Expression type is not assignable to decorator type. Check type of parameters, or missing ().": {

```

Comment:
I would say this is not the right place to fix this issue. in checker.ts in `resolveDecorator` we can check if the signatures did not match, and the decorator node is not a call expression (i.e. `@dec` vs `@dec(..)`) we can add an explanation message to say something like `have you forgot to call it?`.